### ISSUE TITLE
Step 7 error after filtering - IndexError

### ISSUE THREAD

**Issue by isaacovercast** [dereneaton/ipyrad#278]
Looks like it's inside `filter_minsamp`. This is _with_ a pops file. Reported by @Ollie_W_White_twitter on gitter.

From stdout:
```
  Step 7: Filter and write output files for 68 Samples
  [####################] 100%  filtering loci        | 0:00:48
  Encountered an error (see details in ./ipyrad_log.txt)
  Error summary is below -------------------------------
error in filter_stacks on chunk 0: IndexError(index 71 is out of bounds for axis 1 with size 68)
```

From the log file:
```
  [78 78 78 ..., 78 78 78]
  [78 78 78 ..., 78 78 78]]]
2017-10-26 23:39:06,135         pid=19247       [write_outfiles.py]     INFO    passed edges 194494
2017-10-26 23:39:06,135         pid=19247       [write_outfiles.py]     INFO    POPULATIONS {'pop28': (1, ['vi123', 'vi125']), 'pop29': (1, ['we49', 'we50']), 'pop22': (1, ['hi38', 'hi47']), 'pop23': (1, ['li321', 'li325']), 'pop20': (1, ['gr169', 'gr172']), 'pop21': (1, ['hao56', 'hao57']), 'pop26': (1, ['svR119a', 'svR119b']), 'pop27': (1, ['te159', 'te160']), 'pop24': (1, ['ma775', 'ma777']), 'pop25': (1, ['pimoR25', 'pipiR38', 'pisuR3']), 'pop17': (1, ['frfr567', 'frfr585', 'frfr611', 'frfr620']), 'pop16': (1, ['frfo107', 'frfo116']), 'pop15': (1, ['frca319', 'frca320']), 'pop14': (1, ['fo142', 'fo144']), 'pop13': (1, ['fi344', 'fi346']), 'pop12': (1, ['es335', 'es338']), 'pop11': (1, ['diR13', 'diR19']), 'pop10': (1, ['co80', 'coR107b']), 'pop19': (1, ['frsu229', 'frsu234', 'frsu244']), 'pop18': (1, ['frpa101', 'frpa92']), 'pop3': (1, ['addu163', 'addu166']), 'pop2': (1, ['adca363', 'adca366']), 'pop1': (1, ['adad120', 'adad135']), 'pop7': (1, ['brbr157', 'brbr494', 'brbr552', 'brbr664', 'brbr674', 'brbr719', 'brbr749', 'brbr768']), 'pop6': (1, ['adpa58', 'adpa62']), 'pop5': (1, ['adgr356', 'adgr360']), 'pop4': (1, ['ader43', 'ader44']), 'pop9': (1, ['ca95', 'ca97']), 'pop8': (1, ['brgo110', 'brgo112'])}
2017-10-26 23:39:06,330         pid=19217       [write_outfiles.py]     ERROR   error in filter_stacks on chunk 0: IndexError(index 71 is out of bounds for axis 1 with size 68)
2017-10-26 23:39:06,330         pid=19217       [write_outfiles.py]     INFO    finished filtering
2017-10-26 23:39:06,331         pid=19217       [assembly.py]   ERROR   IPyradWarningExit: error in filter_stacks on chunk 0: IndexError(index 71 is out of bounds for axis 1 with size 68)
2017-10-26 23:39:08,429         pid=19217       [assembly.py]   INFO      shutting down engines
```

**Comment by isaacovercast** [comment]
Steps to reproduce:
* Run sim data up to step 6.
* `ipyrad -p params.txt -b subset 1A_0 1B_0 2E_0 2F_0 3I_0 3J_0`
* `ipyrad -p params.txt -s 7`

```
  Step 7: Filter and write output files for 6 Samples                                                                                                                   
  [####################] 100%  filtering loci        | 0:00:04
                                                                                                                                                                        
  Encountered an error (see details in ./ipyrad_log.txt)
  Error summary is below -------------------------------                                                                                                                
error in filter_stacks on chunk 0: IndexError(index 8 is out of bounds for axis 1 with size 6)
```

**Comment by isaacovercast** [comment]
Inside `filter_stacks` the shape of superints is determined by len(sidx), but sidx can contain indices greater than this value.

**Comment by isaacovercast** [comment]
This is fucked. We should really fix it. I think it _only_ happens with the pops file though.

**Comment by jzajdel** [comment]
What ended up being the solution to this problem? I'm currently running version 7.24 and have encountered this exact issue.

**Comment by isaacovercast** [comment]
This only happens if you branch after step 6, remove samples and are using the pops_file to specify minsamples per population, I _think_. Is this the case?

Options:
1) Update to the newest version v0.9 and run again from step 3.
2) Using the v0.7 version create a new branch and remove the samples you don't want, then re-run step 6. Then you can run step 7 and it should work.

Let me know how it goes.
-isaac

**Comment by jzajdel** [comment]
I'm not branching at any point but I am using a pops_file to specify a minsamples per population. This may open up a new issue but I tried using V0.9 and after step 6 I get a error that states that the consensus files cannot be found. Step five runs normally and produces consensus files so I am not sure why this is the case. Preferably, I'd like to keep all the samples. 

Thanks for your Help,

-Josh

**Comment by isaacovercast** [comment]
Hi Josh,
Can you show me the exact error message generated by step 6?

Have you tried running w/o the pops_file as a first pass? This will give
you more information about the average locus coverage of the samples, and
will help inform the settings in the pops_file.

**Comment by jzajdel** [comment]
Hi Issac,

I get the same error on step 7 with version V 0.7 (error in filter_stacks on chunk 0: IndexError(index 1 is out of bounds for axis 0 with size 1) even when not using a pops_file.

For step 6 with version 9 the error I get is;
 ---------------------------------------------------------------------------AssertionError                           Traceback (most recent call last)<string> in <module>
~/miniconda3/lib/python3.7/site-packages/ipyrad/assemble/clustmap_across.py in build_concat_files(data, jobid, samples, randomseed)
    862         sample.stats.reads_consens]
    863     conshandles.sort()
--> 864     assert conshandles, "no consensus files found"
    865 
    866     ## concatenate all of the gzipped consens files
AssertionError: no consensus files found

Any help would be greatly appreciated.

-Josh

**Comment by jzajdel** [comment]
I reran all of the steps after installing version 0.9. After step 6 in this version I get the error;

 -AssertionError Traceback (most recent call last) in
~/miniconda3/lib/python3.7/site-packages/ipyrad/assemble/clustmap_across.py in build_concat_files(data, jobid, samples, randomseed)
862 sample.stats.reads_consens]
863 conshandles.sort()
--> 864 assert conshandles, "no consensus files found"
865
866 ## concatenate all of the gzipped consens files
AssertionError: no consensus files found

I get this error when running with all of my samples or just a subset. I also get this same error if running Version 0.9 on a different computer or on a computing cluster. Step 5 finishes successfully and I can see the resulting consensus files in the directory step 5 creates. I'm really not sure why this error is occuring.

-Josh

**Comment by isaacovercast** [comment]
Can you show me the results from running `ipyrad -p <yourparams.txt> -r`?

Can you show me `ls -ltr` in your consens directory?

Can you show me your params file?

**Comment by isaacovercast** [comment]
This is fixed in v9
